[AMDGPU] Insert `s_wait_xcnt(0)` before atomics to work around write combining misses hazards

shiltian opened the pull request:
This patch adds a workaround for a hazard on GFX1250, which inserts an `s_wait_xcnt(0)` instruction before any atomic operation that might write to memory.

Fixes SWDEV-543703.

rampitec reviewed (commented):
It would be better to do it in the SIInsertWaitcounts. That way you would factor in other waits, so you do not double wait.

shiltian committed:
[AMDGPU] Insert `s_wait_xcnt(0)` before atomics to work around write-combining miss hazard

This patch adds a workaround for a hazzard on GFX1250, which inserts an `s_wait_xcnt(0)` instruction before any atomic operation that might write to memory.

Fixes SWDEV-543703.

shiltian committed:
remove unnecessary `mayStore` check
